hypershift-operator/controllers/nodepool/kubevirt/kubevirt.go

```diff
@@ -445,3 +451,88 @@ func applyJsonPatch(tmplt *[]byte, patch string) error {
 
 	return nil
 }
+
+func ReconcileDefaultIngressEndpoints(ctx context.Context, cli client.Client, hc *hyperv1.HostedCluster, machines []capiv1.Machine) error {
+	var namespace string
+	if hc.Spec.Platform.Kubevirt.Credentials != nil {
+		namespace = hc.Spec.Platform.Kubevirt.Credentials.InfraNamespace
+	} else {
+		namespace = manifests.HostedControlPlaneNamespace(hc.Namespace, hc.Name)
+	}
+
+	// Manifests for infra/mgmt cluster passthrough service
+	cpService := hcpmanifests.IngressDefaultIngressPassthroughService(namespace)
+
+	cpService.Name = fmt.Sprintf("%s-%s",
+		hcpmanifests.IngressDefaultIngressPassthroughServiceName,
+		hc.Spec.Platform.Kubevirt.GenerateID)
+
+	if err := cli.Get(ctx, client.ObjectKeyFromObject(cpService), cpService); err != nil {
+		return fmt.Errorf("failed to get default ingress passthrow Service: %w", err)
+	}
+
+	// If there is a selector endpoints should not be generated
+	if len(cpService.Spec.Selector) > 0 {
+		return nil
+	}
+
+	cpEndpoints := &corev1.Endpoints{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: cpService.Namespace,
+			Name:      cpService.Name,
+		},
+	}
+	machineInternalIPs := []string{}
+	for _, machine := range machines {
+		machineInternalIP := findMachineInternalIP(machine)
+		if machineInternalIP != "" {
+			machineInternalIPs = append(machineInternalIPs, machineInternalIP)
+		}
+	}
```

Comment:
i think we need to consider readiness here of each machine as well. We shouldn't be adding IPs as an endpoint for non-ready machines because it will result in unreachable endpoints.

Reply:
I will add them as `NotReadyAddress`